libcloud/container/drivers/lxd.py

```diff
@@ -476,22 +518,47 @@ def deploy_container(self, name, image, cluster=None,
         to use as basis for limits e.g. "c2.micro"
         :type  ex_instance_type: ``str``
 
+        :param ex_timeout: Timeoute
+        :type  ex_timeout: ``int``
+
         :rtype: :class:`libcloud.container.base.Container`
         """
+
+        if parameters:
+            parameters = json.loads(parameters)
+
+            if "mode" in parameters["source"] and \
```

Comment:
You could simplify it and just do ``if parameters["source"].get("mode", None) == "pull": ...``